src/broadcaster/broadcaster.py

```diff
@@ -12,17 +12,24 @@ class BroadcasterType(Enum):
 
 
 class Broadcaster(ABC):
-    """An abstract class to hide the interface required to notify the rest of the system of airmar input events.
-    Messages are async to prevent them from blocking eachother, since RC input may need a high priority."""
+    """ An abstract class for interfaces required to notify the 
+    rest of the system of input events. """
 
-    @abstractmethod
-    def update_data(self, data=None):
-        """ Updates data dictionary with new dictionary. Data must be packaged into dictionary such that
-        key represents data type, value represents data. """
-        pass
+    def __init__(self):
+        self.data = None
+
+    def publish_dictionary(self, data):
```

Comment:
it would make more sense to split off the setting of
`self.data = data`
into a separate function called update_data... better separation of function